src/extension/background-script/actions/payments/list.ts

```diff
@@ -0,0 +1,65 @@
+import { ConnectorTransaction } from "~/extension/background-script/connectors/connector.interface";
+import state from "~/extension/background-script/state";
+import type { DbPayment, MessagePaymentList, Transaction } from "~/types";
+
+import db from "../../db";
+
+const list = async (message: MessagePaymentList) => {
+  const accountId = state.getState().currentAccountId;
+
+  // TODO: Add pagination instead of limiting to 2121
+  const limit = message?.args?.limit || 2121;
+
+  const connector = await state.getState().getConnector();
+
+  // get IndexDB payments
+  const dbPayments = await db.payments
+    .toCollection()
+    .filter((p) => p.accountId === accountId)
+    .limit(limit)
+    .reverse()
+    .sortBy("createdAt");
+
+  try {
+    const result = await connector.getTransactions();
+
+    // map IndexDB payment data onto ConnectorTransaction
+    const transactions: Transaction[] = result.data.transactions.map(
+      (transaction: ConnectorTransaction) => {
+        const dbPayment = dbPayments.find(
+          (p: DbPayment) => p.preimage === transaction.payment_preimage
+        );
+
+        // @Todo: map transactions correctly
+        // @Todo: how to we use/map "title" and "description" and "memo"
```

Comment:
I think in most cases there is only one field (either memo or description or comment). I think we should try to have a single field if possible? it is called `description` in the Transaction type right now, right?

Reply:
Exactly...

https://github.com/lujakob/lightning-browser-extension/blob/aab05df5550c57b9c771dbe52ba0abab1dd41297/src/types.ts#L577